### PR TITLE
Revert "EID-1583 Revert "EID-1504 Allow eIDAS assertions to be unsign…

### PR DESCRIPTION
**Pull request by JohnWattsGDS**
…ed""

We want eIDAS specific behaviour once again - when validating from an eIDAS flow we want to allow unsigned assertions so this reverts the revert that reverted the original work to provide a validateEidas method and corresponding tests.

### CODE REVIEW

File: saml-lib/src/main/java/uk/gov/ida/saml/core/validation/assertion/AssertionValidator.java [hunk 1]
```diff
@@ -72,6 +87,13 @@ public void validate(
         basicAssertionSubjectConfirmationValidator.validate(assertion.getSubject().getSubjectConfirmations().get(0));
     }
 
+    private void validateSignaturePresent(Signature signature, Assertion assertion) {
+        if (!SamlSignatureUtil.isSignaturePresent(signature)) {
```
Comment (instantiator) [inline saml-lib/src/main/java/uk/gov/ida/saml/core/validation/assertion/AssertionValidator.java]: How about `SamlSignatureUtil.isSignaturePopulated`?

Comment (JohnWattsGDS) [reply saml-lib/src/main/java/uk/gov/ida/saml/core/validation/assertion/AssertionValidator.java]: Yeah, that sounds reasonable.  I'll wait for any other suggestions before changing it.

File: saml-lib/src/test/java/uk/gov/ida/saml/security/SamlAssertionsSignatureValidatorTest.java [hunk 2]
```diff
@@ -52,6 +52,18 @@ public void shouldValidateAllAssertions() {
         verify(samlMessageSignatureValidator).validate(assertion2, IDPSSODescriptor.DEFAULT_ELEMENT_NAME);
     }
 
+    @Test
+    public void shouldValidateEidasAssertionsWithAndWithoutSignature() {
```
Comment (CharlesIC) [inline saml-lib/src/test/java/uk/gov/ida/saml/security/SamlAssertionsSignatureValidatorTest.java]: Better to split this into two tests in another PR
